app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,98 @@
+# rubocop:disable Style/IdenticalConditionalBranches
+
+class FriendshipsController < ApplicationController
+  def create
+    case params[:requestname]
+    when 'Send'
+      send_request
+    when 'Approve'
+      approve_request
+    when 'Unfriend'
+      unfriend_request
+    end
+  end
```

Comment:
Remember the rule thin controllers and thick models. This should be in your friendship model.